### ISSUE TITLE
Memory functions such as M+ M- MRC MC

### ISSUE THREAD

**Issue by anonimuos** [Qalculate/qalculate-gtk#192]
Sorry if I am asking something really stupid or naïve but I didn't find it - can the developer please add memory functions such as M+ M- MRC MC. I know Qalculate has a STO button like TI graphing calculators to assign a result to an arbitrary variable name but memory buttons are handy and useful, almost practically necessary for the simplest day-to-day budgetary or financial calculations.

**Comment by hanna-kn** [comment]
Reasons for the absence of a memory buttons (in addition to the ability to store as variable):
- memory buttons are primarily useful on simple calculators with immediate execution/single-step mode (without support for expressions)
- the ANS button (and the "ans", "ans2", ... variables) can in many cases be used similarly
- the history list can easily be used to recall previous result (using double click or answer() function) or make calculations using  multiple previous values (to some extent the RPN stack can be used similarly)

I've personally almost never used the memory keys on any calculator. It would be helpful if you could describe the kind of calculations where they would be useful.

**Comment by hanna-kn** [comment]
It would be easy to add memory functions, but I need to find a place for them in the GUI.

**Comment by hanna-kn** [comment]
I have now implemented MR (MRC), MC, MS, M+, and M-. They have been added to the "AC", "DEL", "=", and "+" buttons as right-click/middle-click actions. 

It will be possible to create keyboard shortcuts for the memory functions and I'm currently working on making some keypad buttons customizable.

MC, MS, M+, and M- will also be used recognized as commands in the expression entry (and in qalc).

MR (MRC) is implemented as a variable.
